clients/js/src/cipher.ts

```diff
@@ -79,7 +79,11 @@ function formatFailure(fail: CallFailure): string {
 
 export abstract class Cipher {
   public abstract kind: CipherKind;
+  // The public part of the ephemeral key pair, rotated for each epoch.
   public abstract publicKey: Uint8Array;
+  // The private part of the ephemeral key pair, rotated for each epoch.
+  // Used in combination with a peer's public key for encryption/decryption using the X25519-DeoxysII.
```

Comment:
This reads wrong. Client doesn't have access to Sapphire's private ephemeral key. The client only has access to the Sapphire's public key and then the client's own public/private keypair. The latter is ephemeral as well, but is rotated on-demand, no need to be limited by the epoch.

I would say sth like `publicKey` is the Sapphire's public key rotated each epoch. And `ephemeralKey` is the client's keypair for encrypting/decrypting transactions and queries using the X25519-DeoxysII, rotated on-demand.